marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -198,7 +198,12 @@ declare %private function hent:fix-options-for-explorer(
             </search:range>
           </search:constraint>,
           hent:build-sort-operator($sortable-properties, $entity-namespace-map),
-          hent:fix-options-for-explorer($n/node(), $sortable-properties, $entity-namespace-map)
+          hent:fix-options-for-explorer($n/node(), $sortable-properties, $entity-namespace-map),
+          <search:transform-results apply="snippet">
```

Comment:
To avoid errors in the future, I think we need to adjust the first case on search:options to exclude "search:transform-results" if it exists. So this line needs to be adjusted:

    $n/namespace::node(),

That should become (I think, need to test):

    $n/node()[fn:not(self::search:transform-results)]

There's not a test to add because we don't expect the search options generated by ES to return a search:transform-results (though we could add an ML unit test that sends in an options document with an existing transform-results, and then verify that it's discarded in favor of this one). This is just to prevent an error from happening in the future.

Reply:
es.searchOptionsgenerate function generated the search:transform-results and it defaults to empty-snippet. Datahub overrides this by commenting out. Datahub is not excluding the transform results. It looks like the following
`case element(search:transform-results) return <!--<search:transform-results apply="empty-snippet"></search:transform-results>-->`

We should probably change to the following.
`case element(search:transform-results) return (
            <search:transform-results apply="snippet">
                <per-match-tokens>30</per-match-tokens>
                <max-matches>4</max-matches>
                <max-snippet-chars>200</max-snippet-chars>
            </search:transform-results>
      )`

Reply:
The problem with the case on transform-results is that if ES stops adding a transform-results (not likely, but could happen), then our transform-results won't get added. It's safer then to return () for transform-results, and then add our own like you're doing.

Reply:
I will leave with the existing implementation. If transform-results exists, then they are commented out. If they exist, the transform-results are overridden to datahub specific options, which is happening now.